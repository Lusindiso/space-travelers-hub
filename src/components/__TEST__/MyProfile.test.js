import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/index';
import Profile from '../pages/MyProfile';

describe('Testing mission component', () => {
  it('should render correctly Mission component', () => {
    const tree = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly My Mission', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(screen.getByRole('heading', { name: /My Mission/i })).toBeInTheDocument();
  });

  it('should render correctly My Rockets', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(screen.getByRole('heading', { name: /My Rockets/i })).toBeInTheDocument();
  });
});
