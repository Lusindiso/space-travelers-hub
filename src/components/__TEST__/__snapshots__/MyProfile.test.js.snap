// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing mission component should render correctly Mission component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container d-flex justify-content-center my-5"
      >
        <div>
          <h2>
            My Missions
          </h2>
          <div
            class="card mission-card"
          >
            <ul
              class="list-group list-group-flush"
            >
              <p
                class="p-2 text-danger"
              >
                Please join a mission
              </p>
            </ul>
          </div>
        </div>
        <div>
          <h2
            class="ms-5"
          >
            My Rockets
          </h2>
          <div
            class="card rocket-cart ms-5"
          >
            <ul
              class="list-group list-group-flush"
            >
              <p
                class="p-2 text-danger"
              >
                Please reserve a rocket
              </p>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container d-flex justify-content-center my-5"
    >
      <div>
        <h2>
          My Missions
        </h2>
        <div
          class="card mission-card"
        >
          <ul
            class="list-group list-group-flush"
          >
            <p
              class="p-2 text-danger"
            >
              Please join a mission
            </p>
          </ul>
        </div>
      </div>
      <div>
        <h2
          class="ms-5"
        >
          My Rockets
        </h2>
        <div
          class="card rocket-cart ms-5"
        >
          <ul
            class="list-group list-group-flush"
          >
            <p
              class="p-2 text-danger"
            >
              Please reserve a rocket
            </p>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
